### ISSUE TITLE
BatchSpanProcessor exporter not interrupted test is flaky

### ISSUE THREAD

**Issue by mwear** [open-telemetry/opentelemetry-ruby#170]
This test fails intermittently (both in local runs and on CI). We should fix it.

```
  1) Failure:
OpenTelemetry::SDK::Trace::Export::BatchSpanProcessor::export timeout::normally#test_0001_exporter is not interrupted [/home/circleci/project/sdk/test/opentelemetry/sdk/trace/export/batch_span_processor_test.rb:227]:
Expected: :not_interrupted
  Actual: :called
```

**Comment by fbogsany** [comment]
Sigh. This has come up for a [related test](https://github.com/open-telemetry/opentelemetry-ruby/pull/153) before and [was anticipated in the original PR](https://github.com/open-telemetry/opentelemetry-ruby/pull/133#discussion_r340676385).

**Comment by fbogsany** [comment]
💩 fix is to adjust the `exporter_sleeps_for_millis`. Better fixes welcome, though.

**Comment by fhwang** [comment]
Looks like this issue was fixed by deleting the test in 031eb4d1271455775de613cbdc6a10b6b647aa01?
